File: Year-2025 Problem List/March 2025/2529. Maximum Count of Positive Integer and Negative Integer/2529. Maximum Count of Positive Integer and Negative Integer.py
# Problem          : 2529. Maximum Count of Positive Integer and Negative Integer
# Time Complexity  : O(N)
# Python Solution

class Solution:
    def maximumCount(self, nums: List[int]) -> int:
        neg = 0
        pos = 0

        for num in nums:
            if num < 0:
                neg += 1
            if num > 0:
                pos += 1

        return max(pos, neg)

File: Year-2025 Problem List/March 2025/2529. Maximum Count of Positive Integer and Negative Integer/2529. Maximum Count of Positive Integer and Negative Integer.cpp
/*
# Problem          : 2529. Maximum Count of Positive Integer and Negative Integer
# Time Complexity  : O(N)
# C++ Solution
*/

// Brute Force
class Solution {
public:
    int maximumCount(vector<int>& nums) {
        int neg = 0, pos = 0;
        for (auto num : nums) {
            if (num > 0) {
                pos += 1;
            }
            if (num < 0) {
                neg += 1;
            }
        }
        return max(pos, neg);
    }
};